tensorflow/compiler/xla/service/gpu/gpu_executable.cc

```diff
@@ -308,7 +308,8 @@ GpuExecutable::ResolveConstantGlobals(se::Stream* stream) {
 }
 
 StatusOr<se::DeviceMemoryBase> GpuExecutable::BufferForAllocation(
-    absl::Span<ExecutionInput const> arguments,
+    absl::variant<absl::Span<const ShapedBuffer* const>*,
```

Comment:
Does it need to be a pointer to a span and a pointer to a vector? Why not just `variant<span, vector>`?

Reply:
The variant store the object in itself. Like a union.
So this would trigger a copy. I would prefer not trigger a copy here.

Reply:
Span is itself a pointer+length. It does not make sense to take a pointer to a Span. It does not make sense to pass a vector by pointer either: we could pass it either as a Span by value, or as a vector by value.

Reply:
Done. I forgot that we can replace a ptr by a span. Thanks.